### PR TITLE
IA-2968 Add link to entity (if any) on form details page

### PR DESCRIPTION
**Pull request by bramj**
Some improvements on the visual linking of entities with form submissions:

- If a form submission/instance has a `entity_id`, we display the entity details on the submission details page.
- With this in place, there's no reason why we should show a simplified page of a form submission when navigating to it via the entities (e.g. images are not displayed). Instead we can simply link directly to the form.

https://github.com/BLSQ/iaso/assets/2719739/ecdf1d40-8e71-4b15-8c94-65c767c5ab2b

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [x] New translations have been added or updated if new strings have been introduced in the frontend
- ~~My migrations file are included~~
- ~~Are there enough tests~~
- ~~Documentation has been included (for new feature)~~

### CODE REVIEW

**Comment by bramj** [comment]
@beygorghor unassigned you for review for now. I just realized there's some cleanup still to be done on this PR. Apologies, I'll ping you when it's done!

**Comment by bramj** [comment]
@beygorghor Ready for review now :+1:

File: hat/assets/js/apps/Iaso/domains/entities/components/BeneficiaryBaseInfo.tsx [hunk 1]
```diff
@@ -54,30 +66,64 @@ export const BeneficiaryBaseInfo: FunctionComponent<Props> = ({
...
-    );
+
+    if (isFetchingForm || !beneficiary) {
+        return (
+            <Skeleton
+                sx={{ marginBottom: '32px' }}
+                variant="rectangular"
+                width="100%"
+                height="30vh"
+            />
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/entities/components/BeneficiaryBaseInfo.tsx]: ❤️ 
Also you can straight use theme with sx:
marginBottom: theme => theme.spacing(4)

File: hat/assets/js/apps/Iaso/domains/entities/components/BeneficiaryBaseInfo.tsx [hunk 2]
```diff
@@ -54,30 +66,64 @@ export const BeneficiaryBaseInfo: FunctionComponent<Props> = ({
...
+                            <PaperTableRow
+                                label={field.label}
+                                value={field.value}
+                                key={field.key}
+                            />
+                        ))}
+                    </TableBody>
+                </Table>
+            </Box>
+        );
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/entities/components/BeneficiaryBaseInfo.tsx]: i would prefer a separate component then

**Review by beygorghor — APPROVED** [review]
Just a few remarks

You can also make a more detailed skeleton, it looks a bit weird a big rectangle alone

**Pull request merged**